Core/Tracking.lua

```diff
@@ -121,5 +128,94 @@ function Tracking:FindTrackedItem()
 	end
 end
 
+function Tracking:AddTrackedItem(item)
+	Rarity:Debug("AddTrackedItem function called")
+	if not item then -- Make sure the parameter is set.
+		Rarity:Debug("item parameter needs to be set")
+		return
+	end
+	if not item.itemId then -- Make sure the parameter is set.
+		Rarity:Debug("item.itemId parameter needs to be set")
```

Comment:
Technically, you'd have to validate every possible parameter to be sure. There's actually some older code that does this but it hasn't been updated in a while. At some point that could be refurbished and then we may `ValidateItem` wherever needed, but that's really not a priority for now. I'd probably just remove this.

Or do you expect items without an ID to be passed into this? I don't think we can use those anyway.

Reply:
I think Rarity works off the assumption that items are valid once they get past the import stage (for custom items)?
There are a lot of checks in many places, but IMHO validation should be performed once (where needed) only.